widget/diagramwidget/diagram.go

```diff
@@ -13,7 +13,7 @@ import (
 )
 
 // Verify that interfaces are fully implemented
-var _ fyne.Tappable = (*drawingArea)(nil)
+var _ fyne.Tappable = (*DrawingArea)(nil)
```

Comment:
Why did this become a public type, and also a public field on the diagram?

Reply:
Application-level testing needs to send simulated Drag, Mouse, and Tap events to the DrawingArea. I put comments in the code indicating that this is the only reason that this class and its access are public.

Reply:
Why do you need to expose an API for that? See how, for example, the `test` package can `Tap` anything tappable without needing to know its API...